larg4/Core/larg4Main_module.cc

```diff
@@ -219,6 +222,28 @@ void larg4::larg4Main::beginRun(art::Run & r)
   //get the pointer to the User Interface manager
   UI_ = G4UImanager::GetUIpointer();
 
+  // Find the macro (or try to) along the directory path.
+  string macroLocation = "";
+  bool macroWasFound = pathFinder_.find_file(g4MacroFile_, macroLocation);
+  logInfo_ << "Finding path for " << g4MacroFile_ << "...\nSearch "
+           << (macroWasFound ? "successful " : "unsuccessful ")
+           << "and path is: \n" << macroLocation << "\n" << endl;
+
+  // Execute the macro if we were able to find it
+  if (macroWasFound) {
+    // Create the string containing the execution command
+    logInfo_ << "Executing macro: " << g4MacroFile_ << "\n" << endl;
+    string commandToExecute = "/control/execute ";
+    commandToExecute.append(macroLocation);
+    UI_->ApplyCommand(commandToExecute);
+
+  } else {
+    // If it wasn't found...
+    // Leave a message for the user ...
+    logInfo_ << "Unable to find " << g4MacroFile_ << " in the path(s) "
+             << macroPath_ << endl;
+  }
+
```

Comment:
What happens to the workflow if the alternative macro isn't found?  It doesn't look like the module tries to exit so is the default mac file used instead?